base/utf16.jl

```diff
@@ -74,7 +64,7 @@ function encode16(s::AbstractString)
             push!(buf, UInt16(0xd7c0 + (c>>10)))
             push!(buf, UInt16(0xdc00 + (c & 0x3ff)))
         else
-            throw(ArgumentError("invalid Unicode character (0x$(hex(c)) > 0x10ffff)"))
+            throw(UnicodeError(UTF_ERR_INVALID_CHAR, 0, ch))
```

Comment:
Well, that was one that I added myself (my very first PR).  I can put it back in.
(I was thinking of using the same error message for cases like `0xd800 <= ch <= 0xdfff`,
but that's just a false economy)